### PR TITLE
Support for other signal models

### PR DESCRIPTION
**Pull request by TheMagnetar**
**When merged this pull request will:**
- Adds support for other signal models.
- Converts the model in http://acre2.idi-systems.com/wiki/frameworks/custom-signal-processing and labels it as "arcade mode" (boost ranges).

### CODE REVIEW

**Comment by jonpas** [comment]
This still keeps custom signal function possibility, right?

- Some configuration files were committed
- Code style inconsistencies

**Comment by TheMagnetar** [comment]
Yes, that does not change anything. It is backwards compatible.

File: addons/sys_signal/XEH_preInit.sqf [hunk 1]
```diff
@@ -6,10 +6,14 @@ PREP_RECOMPILE_START;
...
 PREP_RECOMPILE_END;
 
+// CBA Settings
+#include "initSettings.sqf"
+
 if (hasInterface) then {
     DGVAR(showSignalHint) = false;
     DGVAR(terrainScaling) = 1;
     DGVAR(omnidirectionalRadios) = 0;
+    DGVAR(signalModel) = SIGNAL_MODEL_LOS_MULTIPATH;
```
Comment (jonpas) [inline addons/sys_signal/XEH_preInit.sqf:13]: Why is this necessary? It will be set with the setting registration.

File: addons/sys_signal/fnc_getSignal.sqf [hunk 2]
```diff
@@ -31,11 +31,13 @@ if (_count == 0) then {
         private _txAntenna = _x;
         {
             private _rxAntenna = _x;
+            private _model = GVAR(signalModel); // TODO: Change models on the fly if compatible (underwater, better frequency matching)
```
Comment (jonpas) [inline addons/sys_signal/fnc_getSignal.sqf:34]: Changing models on the fly happens already, no? Except if you disable that in the setting below.

File: extensions/src/ACRE2Arma/signal/models/Arcade.cpp [hunk 3]
```diff
@@ -0,0 +1,40 @@
+#include "Arcade.hpp"
+#include <glm/geometric.hpp>
+
+acre::signal::model::Arcade::Arcade() : SignalModel()
+{
```
Comment (jonpas) [inline extensions/src/ACRE2Arma/signal/models/Arcade.cpp]: > Opening braces should always be on the same line as their block identifier.

**Review by jonpas — CHANGES_REQUESTED** [review]

File: addons/sys_signal/XEH_preInit.sqf (hunk 1 above)
Comment (TheMagnetar) [reply addons/sys_signal/XEH_preInit.sqf:13]: Never have a variable uninitialised. It is just safer this way.

File: addons/sys_signal/fnc_getSignal.sqf (hunk 2 above)
Comment (TheMagnetar) [reply addons/sys_signal/fnc_getSignal.sqf:34]: This is a note for later on. There may  be models that depend on how the map is processed but right now it is not the case.

File: extensions/src/ACRE2Arma/signal/models/Arcade.cpp (hunk 3 above)
Comment (TheMagnetar) [reply extensions/src/ACRE2Arma/signal/models/Arcade.cpp]: Done it in new files. Changes in older files will be done in separate PRs

File: addons/sys_signal/XEH_preInit.sqf (hunk 1 above)
Comment (jonpas) [reply addons/sys_signal/XEH_preInit.sqf:13]: But it's not uninitialized. If anything, you are changing the default from settings here.

I am not sure at which stage CBA does it, but you might be overwriting user set option.

Comment (TheMagnetar) [reply addons/sys_signal/XEH_preInit.sqf:13]: All other variables are initialised here: Omnidirectional radios, terrain loss scaling (which is set to 1 and later changed to 0.5 by CBA_settings). So it is the right place in my opinion to init all these. Testing shows that as well.

Comment (jonpas) [reply addons/sys_signal/XEH_preInit.sqf:13]: Seems weird and unneeded. :thinking:

Comment (TheMagnetar) [reply addons/sys_signal/XEH_preInit.sqf:13]: Moved CBA settings after the initialisation of the variables.

Comment (jonpas) [reply addons/sys_signal/XEH_preInit.sqf:13]: This makes a lot more sense now! :laughing: :+1:

**Review by jonpas — APPROVED** [review]
